devtron-labs/dashboard
# feat: modified UX for tooltip 'copy to clip-board' on AppDetails page

# Description

Tooltip toggles to "Copy to clipboard" again and remain open until we click somewhere. But it should auto close after copying the name. And for reference UX refer to the copy feature with [Github Interaction](https://github.com/devtron-labs/devtron/branches)

Ticket [AB375](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=375)
Fixes https://github.com/devtron-labs/devtron/issues/1304

## Type of change


# How Has This Been Tested?
- Go to Dashboard > Applications > Select any Deployed App
- Open App Details Tab and scroll to "APPLICATION OBJECTS"
- Select any Replica-set or Pods
- Now try to copy name of any pod, service etc using "copy icon"
- On hover the clipboard icon should change to blue strokes.
- After clicking the clipboard icon, it would be replaced by green tick with the tool tip "Copied" and revert back to previous state after 2sec. 


# Checklist:


<img width="647" alt="Screenshot 2023-02-06 at 1 34 53 PM" src="https://user-images.githubusercontent.com/68757539/216917701-2b56636b-16a8-4380-9e43-f38b3a6a34a3.png">
<img width="1440" alt="Screenshot 2023-02-06 at 1 36 09 PM" src="https://user-images.githubusercontent.com/68757539/216917803-a7709755-d0c6-4c38-8e9e-5cb707a3dc96.png">
